jbpm-task-assigning/jbpm-task-assigning-core/src/test/java/org/jbpm/task/assigning/model/solver/PriorityHelperTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.task.assigning.model.solver;
+
+import java.util.Random;
+import java.util.stream.Stream;
+
+import org.jbpm.task.assigning.model.solver.PriorityHelper;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class PriorityHelperTest {
+
+    @Rule
```

Comment:
nitpicking: `Rule` annotation is no longer supported in jUnit 5 (thinking about future migration).

Reply:
only option I see here is to add a try {} catch {} (or variants) block to emulate what the @Rule already does. But I'm not sure that this is the best strategy. By following this sort of handmade strategy we'll end-up in a situation were different programmers implement different hooks, and this will be even harder to track when we move to jUnit5. I think the best strategy is to apply the same change in all places at the time we get there.

Reply:
The solution is what @triceo proposed below - using `Assertions.assertThatThrownBy` from the AssertJ library:
https://joel-costigliola.github.io/assertj/

Reply:
suggestion from @triceo  was applied